### PR TITLE
introduced XbaseProposalCreator

### PR DESCRIPTION
**Pull request by cdietrich**
introduced XbaseProposalCreator instead of anonymous inner subclass of DefaultProposalCreator to allow better extensibility #365

Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

**Comment by cdietrich** [comment]
@OLibutzki is this what you thought of?

**Comment by OLibutzki** [comment]
@cdietrich, yes, it's possible to extend XbaseProposalCreator and override `apply(IEObjectDescription)` in order to add some special behaviour and call `super.apply(candidate)` as fallback. It's just the same pattern XbaseProposalCreator and DefaultProposalCreator use.

Thanks for "fixing" this that quickly!

File: org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/contentassist/XbaseProposalProvider.java [hunk 1]
```diff
@@ -128,6 +128,155 @@ public boolean apply(IEObjectDescription input) {
 	@Inject
 	private SyntaxFilteredScopes syntaxFilteredScopes;
 	
+	protected class XbaseProposalCreator extends DefaultProposalCreator {
```
Comment (kthoms) [inline org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/contentassist/XbaseProposalProvider.java]: maybe static inner class or not?

Comment (cdietrich) [reply org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/contentassist/XbaseProposalProvider.java]: parent is not static as well and it and the sub use stuff from outer context

Comment (kthoms) [reply org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/contentassist/XbaseProposalProvider.java]: just wanted to make sure that this was a decision that way

**Pull request merged**